src/modules/cache.js

```diff
@@ -0,0 +1,92 @@
+const { cloneDeep } = require('lodash');
+const LRU = require('lru-cache');
+const IPFSUtils = require('../utils/ipfs');
+
+const defaultOptions = {
+    enabled: true,
+    max: 100,
+    maxMarketplaceKeyAge: 3600 * 1000 // 1 hour
+};
+        
+
+class Cache {
+    
+    constructor(chluIpfs, options = {}) {
+        this.chluIpfs = chluIpfs;
+        this.log = this.chluIpfs.logger.debug;
+        this.init(options);
+    }
+
+    isValidityCached(multihash) {
+        IPFSUtils.validateMultihash(multihash);
+        if (this.options.enabled) {
+            this.log('Reading cache for ' + multihash);
+            // only return true if the value in cache was specifically true
+            const val = this.cache.get(multihash) === true;
+            if (val) {
+                this.log('Cache HIT for ' + multihash);
+                return true;
+            } else {
+                this.log('Cache MISS for ' + multihash);
+            }
+        } else {
+            this.log('Skipping cache read for ' + multihash + ': cache disabled');
+        }
+        return false;
+    }
+
+    cacheValidity(multihash) {
+        IPFSUtils.validateMultihash(multihash);
+        if (this.options.enabled) {
+            this.cache.set(multihash, true);
```

Comment:
Good point, if we write to the cache it means we read the same key first, so the last-used timestamp is already updated to "now" and it's useless to rewrite to the cache if it was hit.

I think I should fix it so that the validator calls the cache write only if it performed the full validation.

Reply:
Fixed: the validator will write to the cache only if has performed a full validation